src/collectd-nagios.c

```diff
@@ -593,9 +596,11 @@ int main(int argc, char **argv) {
 
     switch (c) {
     case 'c':
+      strcpy(range_critical_s, optarg);
```

Comment:
Otherwise `char* name = (char *) malloc(strlen(buffer) + 1);` before strcpy. Then we don't assign 1K of memory at startup which rarely gets used.

Reply:
I tried this:
```
    case 'c':
    {
      char* range_critical_s = (char *) malloc(strlen(optarg) + 1);
      strcpy(range_critical_s, optarg);
      parse_range(optarg, &range_critical_g);
      break;
    }
    case 'w':
    {
      char* range_warning_s = (char *) malloc(strlen(optarg) + 1);
      strcpy(range_warning_s, optarg);
      parse_range(optarg, &range_warning_g);
      break;
    }
```
But it results in the following `collectd-nagios` output:
```
CRITICAL: 3 critical, 0 warning, 0 okay | shortterm=0.280000;(null);(null);; midterm=0.330000;(null);(null);; longterm=0.440000;(null);(null);;
```

Reply:
I'm sorry my c skills are very sparse

Reply:
Then locally, don't define+assign, but simply assign:

```c
...
    case 'c':
    {
      range_critical_s = (char *) malloc(strlen(optarg) + 1);
      strcpy(range_critical_s, optarg);
      parse_range(optarg, &range_critical_g);
      break;
    }
    case 'w':
    {
      range_warning_s = (char *) malloc(strlen(optarg) + 1);
      strcpy(range_warning_s, optarg);
      parse_range(optarg, &range_warning_g);
      break;
    }
...
```